connector/connect/client/jvm/src/test/scala/org/apache/spark/sql/StubbingTestSuite.scala

```diff
@@ -0,0 +1,36 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package org.apache.spark.sql
+
+import org.apache.spark.sql.test.RemoteSparkSession
+
+class StubbingTestSuite extends RemoteSparkSession {
+  private def eval[T](f: => T): T = f
+
+  test("capture of to-be stubbed class") {
```

Comment:
This was a working reproduction for issue. Now that `SparkResult` has moved to common, I guess we need to find a new problem...

Reply:
Yeah, the this test case can no longer reproduce the problem. The new test requires a class that only exists in the `connect-client-jvm` module and it needs to be in `src/main` instead of `src/test`.

Reply:
I manually revert the changes from SPARK-44806 based on this PR and verified that the test cases in `SparkSessionE2ESuite` can pass through Maven. However, it would be best if we could find a new target case.

Reply:
I can confirm it is working with Maven, and that it triggers stubbing on the server side.